### PR TITLE
[expo-gl][ncl] add example of using GLView with gesture handler

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

Add example to NCL on how to use GLView with gesture handler and reanimated worklets
Improve expo-gl api when used with worklets

# How

- for now, expo go is using a version of reanimated that is not supported, so the example is still running on js thread
- to use gl in worklet
  - get context id from gl object `const ctx = gl.contextId`
  - pass id via runOnUi function from reanimated
  - get gl object inside worklet code by accessing `GLView.getWorkletContext(contextId)`)
- add example to ncl that is rendering sth based on gesture handler events

docs - https://github.com/expo/expo/pull/15370

# Test Plan

- tested on bare project with latest reanimated (android)
- tested on client with older reanimated (animation running on js thread) (android + ios)

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-gl/src/GLView.tsx [hunk 1]
```diff
@@ -73,6 +73,10 @@ export class GLView extends React.Component<GLViewProps> {
     return ExponentGLObjectManager.takeSnapshotAsync(exglCtxId, options);
   }
 
+  static getWorkletContext(_ctxId: number): ExpoWebGLRenderingContext | undefined {
```
Comment (bbarthec) [inline packages/expo-gl/src/GLView.tsx:76]: You can inline this injection you've proposed by the end of this file using https://developer.mozilla.org/en-US/docs/Glossary/IIFE or am I not understanding the IIFE? 🤔

**Review by bbarthec — COMMENTED** [review]
It looks neat! 😁 Great job on understanding the whole Reanimated/GestureHandler API 👏 
I've got just a simple suggestion/question 😉

File: packages/expo-gl/src/GLView.tsx [hunk 2]
```diff
@@ -73,6 +73,10 @@ export class GLView extends React.Component<GLViewProps> {
...
+    throw new Error('Worklet runtime is not available');
+  }
```
Comment (bbarthec) [inline packages/expo-gl/src/GLView.tsx:78]: By inlining IIFE I meant something like this:
```suggestion
  static getWorkletContext: (
    contextId: number
  ) => ExpoWebGLRenderingContext | undefined = (function () {
    try {
      // reanimated needs to be imported before any workletized code
      // is created, but we don't want to make it dependency on expo-gl.
      require("react-native-reanimated");
      return (contextId: number): ExpoWebGLRenderingContext | undefined => {
        "worklet";
        return global.__EXGLContexts?.[String(contextId)];
      };
    } catch {
      return () => {
        throw new Error("Worklet runtime is not available");
      };
    }
  })();
```

My intention is to keep all the code that is related to this method in one place and not scattered across different parts of this file 😉

**Review by bbarthec — APPROVED** [review]
Awesome work 👏

**Review by tsapeta — APPROVED** [review]

**Pull request merged**